trunk/src/GMP/base/input_DEFAULT.F90

```diff
@@ -221,12 +227,30 @@ subroutine input_DEFAULT(Fp)
       read(nin,*) NRDOMAIN
 !
 !---------------------------------------------------------------------
-!  POINTS                                                            |
+!  POINTS  and surface domains flag                                  |
 !---------------------------------------------------------------------
 !
 !  ...read in number of points
       read(nin,*) NRPOINT
- IF (.NOT. QUIET_MODE)     write(*,1009) NRPOINT
+!
+!  ...most likely, we have read the surface domains flag
+      if ((NRPOINT.eq.0).or.(NRPOINT.eq.1)) then
+        isurf_flag = NRPOINT
+!
+!  .....proceed with reading the actual number of points
+        read(nin,*) NRPOINT
+!
+!  ...most likely, the surface domains flag is missing, set it to 0
+      else
+        IF (.NOT. QUIET_MODE) THEN
```

Comment:
I think this is a bit too verbose. If the user forgot to set the flag but has surface domains specified their mesh will be messed up anyway. IMO would be nicer to simply write the number of distinct surface domains (which in this case is 0).

Reply:
Yes, I'll change that; LD put this in for testing initially.